infra/bicep/scripts/deploy-aks-infra-bicep.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+set -e
+
+script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+# Get the directory of the Helm chart
+HELM_CHART_DIR="$script_dir/../../helm/aoaisim"
+if [[ -f "$script_dir/../../../.env" ]]; then
+	echo "Loading .env"
+	source "$script_dir/../../../.env"
+fi
+
+if [[ ${#BASENAME} -eq 0 ]]; then
+  echo 'ERROR: Missing environment variable BASENAME' 1>&2
+  exit 6
+fi
+
+if [[ ${#LOCATION} -eq 0 ]]; then
+  echo 'ERROR: Missing environment variable LOCATION' 1>&2
+  exit 6
+fi
+
+if [[ ${#SIMULATOR_MODE} -eq 0 ]]; then
+  echo 'ERROR: Missing environment variable SIMULATOR_MODE' 1>&2
+  exit 6
+fi
+
+
+if [[ ${#RECORDING_DIR} -eq 0 ]]; then
+  echo 'Defaulting to /mnt/simulator/recording for RECORDING_DIR' 1>&2
+  RECORDING_DIR="/mnt/simulator/recording"
+fi
+
+# Copy OPENAI_DEPLOYMENT_CONFIG_PATH to Helm Chart
+if [[ ${#OPENAI_DEPLOYMENT_CONFIG_PATH} -eq 0 ]]; then
+  echo 'ERROR: Missing environment variable OPENAI_DEPLOYMENT_CONFIG_PATH' 1>&2
+  exit 6
+fi
+
+cp "$OPENAI_DEPLOYMENT_CONFIG_PATH" "$HELM_CHART_DIR/.openai_deployment_config.json"
+
+USER_ID=$(az ad signed-in-user show --output tsv --query id)
+RESOURCE_GROUP_NAME=${RESOURCE_GROUP_NAME:-aoaisim}
+
+cat << EOF > "$script_dir/azuredeploy.parameters.json"
+{
+  "\$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "location": {
+      "value": "${LOCATION}"
+    },
+    "baseName": {
+      "value": "${BASENAME}"
+    },
+    "simulatorApiKey": {
+      "value": "${SIMULATOR_API_KEY}"
```

Comment:
I have this value set in my `.env` file locally but it isn't being applied to the deployed pod. If I kill the pod and check the logs, I see that a new API Key is created each time (which indicates that the value isn't being set on the pod)

Reply:
I traced this to an issue in the configuration for the Secret Provider. Fix 👍 

![image](https://github.com/user-attachments/assets/6a61c2a0-b8fc-4454-b1a8-982caff7defe)